modules/ROOT/pages/apikit-graphql-api-implementation.adoc

```diff
@@ -0,0 +1,525 @@
+= Querying a GraphQL API Implementation
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+include::reuse::partial$beta-banner.adoc[tag="apikit"]
+
+The following examples showcase the implementation of a GraphQL service to query data using APIkit for GraphQL.
+
+Before running these examples, download and run the *apikit-graphql-example* application, which loads a sample dataset in memory for testing purposes. Get more information about https://graphql.org/learn/serving-over-http/[the format of the queries sent through an HTTP POST request^].
+
+The example application is based on the following GraphQL API:
+
+----
+type Post {
+    id: ID!
+    user: User
+    title: String
+    body: String
+    comments: [Comment]
+}
+
+type Comment {
+    id: ID!
+    user: User
+    post: Post
+    body: String
+    replies: [Comment]
+}
+
+type User {
+    id: ID!
+    name: String
+    username: String
+    email: String
+    phone: String
+    website: String
+    address: Address
+}
+
+type Address {
+    id: ID!
+    street: String
+    city: String
+    zipcode: String
+    latitude: Float
+    longitude: Float
+}
+
+input AddressInput {
+    street: String
+    city: String
+    zipcode: String
+    latitude: Float
+    longitude: Float
+}
+
+input UserInput {
+    name: String
+    username: String
+    email: String
+    phone: String
+    website: String
+    address: AddressInput
+}
+
+type Query {
+    # Looks for a user by ID
+    getUser(id: ID!): User
+    # Looks for a post by ID
+    getPost(id: ID!): Post
+    # Returns the list of all users
+    allUsers: [User]
+    # Returns the list of all posts
+    allPosts: [Post]
+}
+
+type Mutation {
+    # Creates a new user
+    createUser(user: UserInput!): User!
+    # Creates a new post
+    createPost(userId: ID!, title: String!, body: String!): Post!
+    # Adds a comment to a post
+    postComment(userId: ID!, postId: ID!, body: String!): Comment!
+    # Adds a reply to a comment
+    replyToComment(commentId: ID!, userId: ID!, body: String!): Comment!
+}
+----
+
+== Request Queries Examples
+
+In the following example, the type *Query* defines the list of queries that the service exposes.
+
+----
+...
+type Query {
+    # Looks for a user by ID
+    getUser(id: ID!): User
+    # Looks for a post by ID
+    getPost(id: ID!): Post
+    # Returns the list of all users
+    allUsers: [User]
+    # Returns the list of all posts
+    allPosts: [Post]
+}
+...
+----
+
+=== Retrieve All the Posts
+
+This use case makes a simple request for the data of each existing post.
+ 
+==== Make a Query
```

Comment:
Is Make the right word here?

Reply:
It has been decided to keep "Make a Query". Thanks!